### ISSUE TITLE
`no-unnecessary-splice`: false positive when the result of `splice(0)` is used

### ISSUE THREAD

**Issue by ocavue** [sindresorhus/eslint-plugin-unicorn#3610]
### Description

`eslint-plugin-unicorn@73.0.0`, `eslint@10.8.0`

```js
// ❌ error  Prefer setting `.length = 0` over `.splice()` to empty an array  unicorn/no-unnecessary-splice
// disposers: Array<() => void>
for (const teardown of disposers.splice(0)) {
  teardown()
}
```
 

The removed elements are being used, so this is not (only) emptying an array, it is an atomic "read all, then empty".

The only behavior-preserving rewrite is longer than the original, which contradicts the rule's description ("Disallow `Array#splice()` when simpler alternatives exist"):

```js
// ✅
const pending = [...disposers]
disposers.length = 0
for (const teardown of pending) {
  teardown()
}
```

**Expected:** no report for `splice(0)` when the return value is used.

**Possible patch:** Add the following code at https://github.com/sindresorhus/eslint-plugin-unicorn/blob/2e19a9401d84cc0d2f7f73acc33b4e4f154685de/rules/no-unnecessary-splice.js#L126

```ts
// `array.splice(0)` whose result is used is an atomic "read all, then empty".
// `.length = 0` evaluates to `0`, so there is no simpler alternative here.
if (replacement.messageId === MESSAGE_ID_EMPTY && !isValueNotUsable(callExpression)) {
    return;
}
```

### Examples

(See above)
